Dockerfile

```diff
@@ -9,7 +9,18 @@ ENV QT_X11_NO_MITSHM=1
 RUN apt-get update && \
     apt-get install python3-catkin-tools -y
 
-RUN source /opt/ros/noetic/setup.bash
+RUN echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc
+
+# Install pigpio
+RUN apt-get install -y wget unzip && \
+    wget https://github.com/joan2937/pigpio/archive/master.zip && \
+    unzip master.zip && \
+    cd pigpio-master && \
+    make && \
+    make install && \
+    cd .. && \
+    rm -rf pigpio-master && \
+    rm master.zip
```

Comment:
Here you are installing the C library which is necessary in the raspberry but not in the docker image.

I would suggest installing the python lib instead.

```suggestion
# Install pip tools 
RUN apt-get install -y python3-pip

# Install pigpio python module
RUN pip3 install pigpio
```

Reply:
I did not know, so I think it is better with just that command